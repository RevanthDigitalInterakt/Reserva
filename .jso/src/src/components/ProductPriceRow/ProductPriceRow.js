  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[1]);
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function ProductPriceRow(_ref) {
    var installments = _ref.installments,
      currency = _ref.currency,
      discountTag = _ref.discountTag,
      priceWithDiscount = _ref.priceWithDiscount,
      price = _ref.price;
    var finalPrice = (0, _react.useMemo)(function () {
      return discountTag && priceWithDiscount ? priceWithDiscount : price;
    }, [discountTag, priceWithDiscount, price]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsxs)(_reactNative.View, {
      style: {
        flexDirection: 'row',
        alignItems: 'center'
      },
      children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsx)(_reactNative.Text, {
        style: _$$_REQUIRE(_dependencyMap[3]).styles.textPrice,
        children: `${currency} `
      }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsx)(_reactNative.Text, {
        style: _$$_REQUIRE(_dependencyMap[3]).styles.textPrice,
        children: `${(0, _$$_REQUIRE(_dependencyMap[4]).integerPart)(finalPrice)},`
      }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsx)(_reactNative.Text, {
        style: _$$_REQUIRE(_dependencyMap[3]).styles.textPrice,
        children: `${(0, _$$_REQUIRE(_dependencyMap[4]).decimalPart)(finalPrice)}`
      }), !!(installments != null && installments.number) && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsxs)(_$$_REQUIRE(_dependencyMap[2]).Fragment, {
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsx)(_reactNative.View, {
          style: _$$_REQUIRE(_dependencyMap[3]).styles.divider
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsxs)(_reactNative.Text, {
          style: [_$$_REQUIRE(_dependencyMap[3]).styles.textInstallments, {
            fontSize: 12
          }],
          children: [installments.number, "x"]
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsx)(_reactNative.Text, {
          style: _$$_REQUIRE(_dependencyMap[3]).styles.textInstallments,
          children: ` ${currency} ${(0, _$$_REQUIRE(_dependencyMap[4]).integerPart)(installments.value)},`
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsx)(_reactNative.Text, {
          style: _$$_REQUIRE(_dependencyMap[3]).styles.textInstallments,
          children: `${(0, _$$_REQUIRE(_dependencyMap[4]).decimalPart)(installments.value)}`
        })]
      })]
    });
  }
  var _default = exports.default = ProductPriceRow;
