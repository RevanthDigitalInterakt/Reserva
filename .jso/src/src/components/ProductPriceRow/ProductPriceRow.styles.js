  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.styles = undefined;
  var _reactNative = _$$_REQUIRE(_dependencyMap[0]);
  var _configDeviceSizes = _$$_REQUIRE(_dependencyMap[1])(_$$_REQUIRE(_dependencyMap[2]));
  var styles = exports.styles = _reactNative.StyleSheet.create({
    textPrice: {
      fontFamily: 'ReservaSans-Medium',
      fontSize: 12,
      color: '#8A8C8E',
      letterSpacing: 0.05
    },
    grey: {
      color: _$$_REQUIRE(_dependencyMap[3]).COLORS.LIGHT_GRAY
    },
    black: {
      color: _$$_REQUIRE(_dependencyMap[3]).COLORS.BLACK
    },
    textInstallments: {
      fontFamily: 'ReservaSans-Medium',
      fontSize: 12,
      color: '#333333',
      letterSpacing: 0.05
    },
    divider: {
      backgroundColor: '#8A8C8E',
      width: 1,
      height: 16,
      marginHorizontal: _configDeviceSizes.default.DEVICE_WIDTH > 320 ? 6 : 4
    },
    textPricePrime: {
      fontFamily: 'ReservaSans-Medium',
      fontSize: 12,
      color: _$$_REQUIRE(_dependencyMap[3]).COLORS.RED,
      letterSpacing: 0.05
    },
    containerPrime: {
      padding: 2,
      backgroundColor: _$$_REQUIRE(_dependencyMap[3]).COLORS.RED,
      marginLeft: 10
    },
    labelPrime: {
      fontFamily: 'ReservaDisplay-Regular',
      color: _$$_REQUIRE(_dependencyMap[3]).COLORS.WHITE,
      fontSize: 8
    },
    flex: {
      flexDirection: 'row',
      alignItems: 'center'
    }
  });
