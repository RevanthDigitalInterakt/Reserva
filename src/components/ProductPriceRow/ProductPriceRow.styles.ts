import { StyleSheet } from 'react-native';
import configDeviceSizes from '../../utils/configDeviceSizes';

export const styles = StyleSheet.create({
  textPrice: {
    fontFamily: 'ReservaSans-Medium',
    fontSize: 12,
    color: '#8A8C8E',
    letterSpacing: 0.05,
  },
  textInstallments: {
    fontFamily: 'ReservaSans-Medium',
    fontSize: 12,
    color: '#333333',
    letterSpacing: 0.05,
  },
  divider: {
    backgroundColor: '#8A8C8E',
    width: 1,
    height: 16,
    marginHorizontal: configDeviceSizes.DEVICE_WIDTH > 320 ? 6 : 4,
  },
});
