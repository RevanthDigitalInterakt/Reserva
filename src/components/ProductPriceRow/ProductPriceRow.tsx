import React, { useMemo } from 'react';
import { Text, View } from 'react-native';
import { integerPart, decimalPart } from '../../utils/numberUtils';
import { styles } from './ProductPriceRow.styles';

interface IPros {
  installmentsNumber: number;
  currency?: string;
  installmentsPrice: number
  discountTag?: number;
  priceWithDiscount?: number;
  price: number;
}

const ProductPriceRow = ({
  installmentsNumber,
  currency,
  installmentsPrice,
  discountTag,
  priceWithDiscount,
  price,
}: IPros) => {
  const finalPrice = useMemo(() => (
    discountTag && priceWithDiscount ? priceWithDiscount : price
  ), [discountTag, priceWithDiscount, price]);

  return (
    <View style={{ flexDirection: 'row', alignItems: 'center' }}>

      <Text style={styles.textPrice}>
        {`${currency} `}
      </Text>

      <Text style={styles.textPrice}>
        {`${integerPart(finalPrice)},`}
      </Text>

      <Text style={styles.textPrice}>
        {`${decimalPart(finalPrice)}`}
      </Text>

      <View style={styles.divider} />

      <Text style={[styles.textInstallments, { fontSize: 12 }]}>
        {installmentsNumber}
        x
      </Text>
      <Text style={styles.textInstallments}>
        {` ${currency} ${integerPart(installmentsPrice)},`}
      </Text>
      <Text style={styles.textInstallments}>
        {`${decimalPart(installmentsPrice)}`}
      </Text>
    </View>
  );
};

export default ProductPriceRow;
