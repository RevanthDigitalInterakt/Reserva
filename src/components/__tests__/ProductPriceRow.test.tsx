import React from 'react';
import {
  render,
} from '@testing-library/react-native';
import ProductPriceRow from '../ProductPriceRow/ProductPriceRow';

describe('ProductPriceRow', () => {
  it('should match with the snapshot', async () => {
    const Component = (
      <ProductPriceRow
        installmentsNumber={6}
        installmentsPrice={200}
        discountTag={100}
        priceWithDiscount={150}
        price={1000}
      />
    );

    const { toJSON } = render(Component);

    expect(toJSON()).toMatchSnapshot();
  });
});
