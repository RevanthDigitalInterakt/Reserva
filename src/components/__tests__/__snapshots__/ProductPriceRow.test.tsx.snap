// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPriceRow should match with the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      {
        "color": "#8A8C8E",
        "fontFamily": "ReservaSans-Medium",
        "fontSize": 12,
        "letterSpacing": 0.05,
      }
    }
  >
    undefined 
  </Text>
  <Text
    style={
      {
        "color": "#8A8C8E",
        "fontFamily": "ReservaSans-Medium",
        "fontSize": 12,
        "letterSpacing": 0.05,
      }
    }
  >
    150,
  </Text>
  <Text
    style={
      {
        "color": "#8A8C8E",
        "fontFamily": "ReservaSans-Medium",
        "fontSize": 12,
        "letterSpacing": 0.05,
      }
    }
  >
    00
  </Text>
</View>
`;
